client/src/components/App.js

```diff
@@ -12,27 +12,21 @@ class App extends Component {
       modal: {type: '', active_project: {}},
     };
     this.showProjects = this.showProjects.bind(this);
-    this.showForm = this.showForm.bind(this);
-    this.hideForm = this.hideForm.bind(this);
-    this.submitNewProject = this.submitNewProject.bind(this);
-    this.submitEditedProject = this.submitEditedProject.bind(this);
-    this.inspectProject = this.inspectProject.bind(this);
+    this.showModal = this.showModal.bind(this);
+    this.hideModal = this.hideModal.bind(this);
+    this.submitProject = this.submitProject.bind(this);
     this.parseProjectForm = this.parseProjectForm.bind(this);
   }
 
   UNSAFE_componentWillMount() {
     this.showProjects();
   }
 
-  inspectProject(active_project) {
-    this.setState({modal: {type: 'details', active_project}});
+  showModal(type, active_project) {
+    this.setState({modal: {type: type, active_project}});
```

Comment:
Could we simplify this to `{type, active_project}`?

Reply:
oh, yes, of course!